From 23d09a547d983dba2c0edcd3df5125128e017cc0 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Wed, 29 Jul 2020 12:02:56 +0530
Subject: [PATCH 16/16] Set up S/PDIF in/out ports

- Add SPDIF nodes
- Also enable S/PDIF-out.
- Create different pinmux setting.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index e010c71197c0..86bd50621377 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -214,6 +214,14 @@
 		hdmi-controller = <&hdmi_audio>;
 	};
 
+	sound-spdif {
+		compatible = "fsl,imx-audio-spdif";
+		model = "imx-spdif";
+		spdif-controller = <&spdif>;
+		spdif-in;
+		spdif-out;
+	};
+
 	mxcfb1: fb@0 {	/* IPU1 DI1: /dev/fb0=HDMI, fb1=IPU1's overlay */
 		compatible = "fsl,mxc_sdc_fb";
 		disp_dev = "hdmi";
@@ -1080,6 +1088,13 @@
 			>;
 		};
 
+		pinctrl_spdif: spdifgrp {
+			fsl,pins = <	/* SMARC SPDIF_IN/OUT */
+				MX6QDL_PAD_GPIO_16__SPDIF_IN            0x1b0b0
+				MX6QDL_PAD_GPIO_19__SPDIF_OUT           0x1b0b0
+			>;
+		};
+
 		pinctrl_uart1: uart1grp {
                        fsl,pins = <    /* SMARC SER0 */
 				MX6QDL_PAD_SD3_DAT6__UART1_RX_DATA      0x1b0b1
@@ -1264,6 +1279,15 @@
 	status = "okay";
 };
 
+&spdif {	/* SMARC SPDIF_IN/OUT */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spdif>;
+	assigned-clocks = <&clks IMX6QDL_CLK_SPDIF_SEL>, <&clks IMX6QDL_CLK_SPDIF_PODF>;
+	assigned-clock-parents = <&clks IMX6QDL_CLK_PLL3_PFD3_454M>;
+	assigned-clock-rates = <0>, <227368421>;
+	status = "okay";
+};
+
 #if CONFIG_I2S_AUDIO
 &ssi2 {	/* SMARC I2S0 (instead of SPI0) */
 	status = "okay";
-- 
2.17.1

